scripts/exglobal_atmos_products.sh

```diff
@@ -148,13 +158,23 @@ for (( nset=1 ; nset <= downset ; nset++ )); do
   # We are in a loop over downset, save output from mpmd into nset specific output
   mv mpmd.out "mpmd_${nset}.out"
 
+  # ---- FIX 3a: default nproc_used when every rank produced work ----
+    : "${nproc_used:=${nproc}}"
   # Concatenate grib files from each processor into a single one
   # and clean-up as you go
   echo "Concatenating processor-specific grib2 files into a single product file"
   for (( iproc = 1 ; iproc <= nproc ; iproc++ )); do
     for grid in "${grids[@]}"; do
-      cat "pgb2${grp}file_${fhr3}_${iproc}_${grid}" >> "pgb2${grp}file_${fhr3}_${grid}"
-      rm  -f "pgb2${grp}file_${fhr3}_${iproc}_${grid}"
+   # ---- FIX 3b: only append files that exist and are non-empty ----
+#      cat "pgb2${grp}file_${fhr3}_${iproc}_${grid}" >> "pgb2${grp}file_${fhr3}_${grid}"
+#      rm  -f "pgb2${grp}file_${fhr3}_${iproc}_${grid}"
```

Comment:
```suggestion
    # only append files that exist and are non-empty
```
Check indent here after commit

Reply:
Updated